sesman.el

```diff
@@ -919,8 +942,12 @@ buffers."
 (defvar sesman--path-cache (make-hash-table :test #'equal))
 ;; path caching because file-truename is very slow
 (defun sesman--expand-path (path)
-  (or (gethash path sesman--path-cache)
-      (puthash path (file-truename path) sesman--path-cache)))
+  (if (or (eq sesman-follow-symlinks t)
+          (and (eq sesman-follow-symlinks 'auto)
+               (sesman--follow-symlink-p path)))
```

Comment:
Here it should be true name instead of `path`. I think this caching of file-names is not necessary either. There is a buffer local variable `buffer-file-truename` which could be used instead.

Reply:
>  think this caching of file-names is not necessary either. 

You mean basically remove `sesman--expand-path` and use something like `(file-name-directory buffer-file-truename)` ?

Reply:
Not sure about [`(sesman--expand-path dir)`](https://github.com/vspinu/sesman/blob/master/sesman.el#L953)

Reply:
> The sesman-relevant-context-p for project should first sesman--expand-path on buffer true name, then use that as default-directory and call sesman-project and root-dir.

@vspinu  I don't think that I have sesman expertise required for a refactoring of such scale :smiley_cat:

Reply:
Sure. Let me take over from here. This part turned to be much stuffier than expected. Thanks for all the effort!

Reply:
@vspinu  Cool thx :+1: